std/format.d

```diff
@@ -5796,6 +5823,79 @@ if (isSomeString!(typeof(fmt)))
     static assert(!__traits(compiles, {s = format!"%d"(4.03);})); // incompatible arg
 }
 
+// called during compilation to guess the length of the
+// result of format
+private size_t guessLength(alias fmtString, Char)()
```

Comment:
In guessLength is there a reason to prefer `alias fmtString ` over making fmtString a function parameter? It seems to me that `guessLength(FmtStringType, Char)(FmtStringType fmtString)` would result in fewer template instantiations.

Reply:
That's a good point. We don't need `guessLength` to be re-instantiated per format string. Since it's going to be run in CTFE anyway, making the format string a "runtime" parameter should work just as well, and likely faster / less memory hogging, given the way dmd currently handles templates.